src/drivers/gitea.go

```diff
@@ -0,0 +1,195 @@
+	if err != nil {
+		return "", err
+	}
+	commitMessageParts := strings.SplitN(options.CommitMessage, "\n", 2)
+	commitTitle := commitMessageParts[0]
+	commitMessage := ""
+	if len(commitMessageParts) == 2 {
+		commitMessage = commitMessageParts[1]
+	}
+	return d.apiMergePullRequest(options.PullRequestNumber, commitTitle, commitMessage)
+}
+
+// Helper
+
+func (d *giteaCodeHostingDriver) connect() {
+	if d.client == nil {
+		ts := oauth2.StaticTokenSource(
+			&oauth2.Token{AccessToken: d.apiToken},
+		)
+		tc := oauth2.NewClient(context.Background(), ts)
+		d.client = gitea.NewClientWithHTTP(fmt.Sprintf("https://%s", d.hostname), tc)
+	}
+}
+
+func getDefaultCommitMessage(pullRequest *gitea.PullRequest) string {
+	return fmt.Sprintf("%s (#%d)", pullRequest.Title, pullRequest.Index)
+}
+
+func filterPullRequests(pullRequests []*gitea.PullRequest, baseName, headName string) []*gitea.PullRequest {
+	pullRequestsFiltered := []*gitea.PullRequest{}
+	for _, pullRequest := range pullRequests {
+		if pullRequest.Base.Name != baseName {
+			break
+		}
+		if headName != "" && pullRequest.Head.Name != headName {
+			break
+		}
+		pullRequestsFiltered = append(pullRequestsFiltered, pullRequest)
+	}
+	return pullRequestsFiltered
+}
+
+func (d *giteaCodeHostingDriver) apiMergePullRequest(pullRequestNumber int64, commitTitle, commitMessage string) (mergeSha string, err error) {
+	_, err = d.client.MergePullRequest(d.owner, d.repository, pullRequestNumber, gitea.MergePullRequestOption{
+		Style:   gitea.MergeStyleSquash,
+		Title:   commitTitle,
+		Message: commitMessage,
+	})
+	if err != nil {
+		return "", err
+	}
+	pullRequest, err := d.client.GetPullRequest(d.owner, d.repository, pullRequestNumber)
+	if err != nil {
+		return "", err
+	}
+	return *pullRequest.MergedCommitID, nil
+}
+
+// retargetPullRequests retargets pullrequests onto a new base branch
+// this comes in handy when an ancestor got merged, so that children can be retargeted to the ancestor's own target branch
+// example:
+//   ancerstor -> master
+//   children1 -> ancestor  --> master (retargeted to master after merge)
+//   children2 -> ancestor  --> master (retargeted to master after merge)
+//nolint:unparam
+func (d *giteaCodeHostingDriver) apiRetargetPullRequests(pullRequests []*gitea.PullRequest, newBaseName string) error {
+	for _, pullRequest := range pullRequests {
+		// if options.LogRequests {
+		// 	helpers.PrintLog(fmt.Sprintf("Gitea API: Updating base branch for PR #%d to #%s", *pullRequest.Index, newBaseName))
+		// }
+		helpers.PrintLog(fmt.Sprintf("Gitea API: Updating base branch for PR #%d to #%s", 1, newBaseName))
+		helpers.PrintLog(fmt.Sprintf("The Gitea API currently does not support retargeting, please restarget #%d manually, see https://github.com/go-gitea/gitea/issues/11552", pullRequest.Index))
+		// _, err = d.client.EditPullRequest(d.owner, d.repository, *pullRequest.Index, &gitea.EditPullRequestOption{
```

Comment:
Can you remove this commented out code? If you want to keep it, please add a comment that explains why it is commented out and when it can be uncommented.